### ISSUE TITLE
Incompatible image size with RandomGeoSampler

### ISSUE THREAD

**Issue by hfangcat** [microsoft/torchgeo#1778]
### Discussed in https://github.com/microsoft/torchgeo/discussions/1773

<div type='discussions-op-text'>

<sup>Originally posted by **hfangcat** December 14, 2023</sup>
The code example is here:
```python
from torchgeo.datasets import stack_samples

from torch.utils.data import DataLoader

from torchgeo.samplers import RandomGeoSampler, RandomBatchGeoSampler
from torchgeo.datasets import ChesapeakeCVPR

root = '/scratch/local/cvpr_chesapeake_landcover/'
dataset_train = ChesapeakeCVPR(root, splits=['md-train'], layers=['naip-new', 'lc', 'nlcd'], download=False)

sampler_train = RandomGeoSampler(dataset_train, size=224, length=1)
loader_train = DataLoader(dataset_train, sampler=sampler_train, collate_fn=stack_samples)

for i, data in enumerate(loader_train):
    images = data["image"]
    targets = data["mask"]
    print(images.shape, targets.shape)
    break
```
Then it shows the shape of images and targets:
```
torch.Size([1, 4, 176, 176]) torch.Size([1, 2, 176, 176])
```
which is not compatible with 224: what I specified in the code `sampler_train = RandomGeoSampler(dataset_train, size=224, length=1)`, can the community help me find out why it happens?</div>

**Comment by dmeaux** [comment]
Since you're just looking at Maryland, have you tried using `ChesapeakeMD` since it inherits from `Chesapeake`, which inherits from `RasterDataset`? I would think that each state's own class would give you a consistent projected resolution.

**Comment by hfangcat** [comment]
> Since you're just looking at Maryland, have you tried using `ChesapeakeMD` since it inherits from `Chesapeake`, which inherits from `RasterDataset`? I would think that each state's own class would give you a consistent projected resolution.

Thanks! Actually, I will use all datasets afterwards for training, this is just an example to test the dataloader :D But I can try what you mentioned if it is really hard to fix

**Comment by adamjstewart** [comment]
Note that `ChesapeakeMD` and friends only have masks, not images. Only `ChesapeakeCVPR` has both images and masks.

**Comment by calebrob6** [comment]
Note, this could be solved if we create the concept of TileDatasets (being discussed in https://github.com/microsoft/torchgeo/pull/1353).

**Comment by adamjstewart** [comment]
Or if we convert ChesapeakeCVPR to a RasterDataset.
